localgov_subsites_extras.api.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Hooks provided by the Locagov Subsites Extras module.
+ */
+
+use Drupal\node\NodeInterface;
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Alter the current node, before subsite determination.
+ *
+ * This alter hook is called before we walk the menu tree to determine if we're
+ * looking at a page in a subsite or not.
+ *
+ * It'll be passed the current node, acquired from routeMatch. If no node is
+ * found there, this hook will be passed null. This gives the opportunity to
+ * include non-node pages in a subsite if you wish.
+ *
+ * If you want to indicate that a subsite is not being viewed, set $node to
+ * NULL.
+ *
+ * Usually, this hook will be used to use a property of the node passed, such as
+ * a reference to another node (such as the parent node field for guides,
+ * directories, etc) to swop out the current node for a different node, which is
+ * in the menu and therefore part of a subsite.
+ */
+function hook_subsites_extra_current_node_alter(?NodeInterface &$node) {
```

Comment:
The hook should match the module name, so should have an extra _s_
`function hook_subsites_extras_current_node_alter(?NodeInterface &$node) {`
Also update the implementations in the module file.

Reply:
It think it should be the module machine name so `hook_localgov_subsites_extras_current_node_alter`
Are there any hooks in other LGD modules?

Reply:
> Are there any hooks in other LGD modules?

Only `hook_localgov_microsites_roles_default` I think.

Reply:
Publications has hook_localgov_publications_menu_tree_alter.

Reply:
I've added the missing s and the localgov_ prefix.